### PR TITLE
[docs] adds intercom to integration

### PR DESCRIPTION
**Pull request by ritwickrajmakhal**
## Description
Adds documentation for intercom handler.
## Type of change

- [x] 📄 This change requires a documentation update

## Additional Media:
image

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: docs/mint.json [hunk 1]
```diff
@@ -70,6 +70,7 @@
     {"source": "/app-integrations/shopify","destination": "/integrations/app-integrations/shopify"},
     {"source": "/app-integrations/twitter","destination": "/integrations/app-integrations/twitter"},
     {"source": "/app-integrations/youtube","destination": "/integrations/app-integrations/youtube"},
+    {"source": "/app-integrations/intercom","destination": "/integrations/app-integrations/intercom"},
```
Comment (martyna-mindsdb) [inline docs/mint.json:73]: It is not required to add it to the redirects list.

Please remove this line.

File: docs/integrations/app-integrations/intercom.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+To create a database using the Intercom engine, you can use a SQL-like syntax as shown below:
+
+```sql
+CREATE DATABASE myintercom
+WITH
+  ENGINE = 'intercom',
+  PARAMETERS = {
+    "access_token" : "your-intercom-access-token"
+  };
+```
```
Comment (martyna-mindsdb) [inline docs/integrations/app-integrations/intercom.mdx:25]: Please move the below content to the end of the `## Connection` section.
```
To create a database using the Intercom engine, you can use a SQL-like syntax as shown below:

```sql
CREATE DATABASE myintercom
WITH
  ENGINE = 'intercom',
  PARAMETERS = {
    "access_token" : "your-intercom-access-token"
  };
```
```

File: docs/integrations/app-integrations/intercom.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### SELECT
```
Comment (martyna-mindsdb) [inline docs/integrations/app-integrations/intercom.mdx:27]: Please remove these section titles: `### SELECT`, `### WHERE`, `### INSERT`, `### UPDATE`.

Just keep the content under `## Usage` section.

**Comment by martyna-mindsdb** [comment]
@ritwickrajmakhal 
Please address the review comments above.

**Comment by ritwickrajmakhal** [comment]
Changes made @martyna-mindsdb

File: docs/integrations/app-integrations/intercom.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+sidebarTitle: Intercom
+---
+
+[Intercom](https://intercom.com) is a software company that provides customer messaging and engagement tools for businesses. They offer products and services for customer support, marketing, and sales, allowing companies to communicate with their customers through various channels like chat, email, and more.
+
+## Connection
+
+To get started with the Intercom API, you need to initialize the API handler with the required access token for authentication. You can do this as follows:
+
+- `access_token`: Your Intercom access token for authentication.
```
Comment (martyna-mindsdb) [inline docs/integrations/app-integrations/intercom.mdx:12]: Please add a link to instructions on how to get the Intercom access token.

**Comment by martyna-mindsdb** [comment]
Thanks @ritwickrajmakhal 

Please address the review comment above - after that, it's ready to go.

**Comment by ritwickrajmakhal** [comment]
Changes made @martyna-mindsdb

**Review by martyna-mindsdb — APPROVED** [review]

**Pull request merged**